include/pmacc/static_assert.hpp

```diff
@@ -45,7 +45,7 @@ namespace pmacc
  * @param pmacc_unique_id pre compiler unique id
  * @param pmacc_typeInfo a type that is shown in error message
  */
-#if ( PMACC_CUDA_COMPILER_CLANG == 1 )
+#if BOOST_LANG_CUDA && BOOST_COMP_CLANG_CUDA
```

Comment:
@BenjaminW3 I think alpaka should either set this define with an `ALPAKA_` prefix or we move it upstream into boost for real